src/TSTransformer/util/assignment.ts

```diff
@@ -21,13 +21,16 @@ const COMPOUND_OPERATOR_MAP = new Map<ts.SyntaxKind, luau.AssignmentOperator>([
 ]);
 
 export function getSimpleAssignmentOperator(
+	state: TransformState,
 	leftType: ts.Type,
 	operatorKind: ts.AssignmentOperator,
 	rightType: ts.Type,
+	node: ts.BinaryExpression,
 ) {
 	// plus
 	if (operatorKind === ts.SyntaxKind.PlusEqualsToken) {
-		return isDefinitelyType(leftType, isStringType) || isDefinitelyType(rightType, isStringType) ? "..=" : "+=";
+		const isString = isDefinitelyType(state, [leftType, rightType], node, isStringType);
```

Comment:
Is this the only use of the array form for `isDefinitelyType`? IMO, this is less readable than before.

Reply:
It might be less readable but has distinctly different behaviour. It short circuits if either type matches internally, and can then skip the any checks for the others.

Reply:
The alternative of doing that behaviour outside isDefinitelyType is somewhere as a codeblock on discord, but it looks even less readable

Reply:
If it is the case that we need this functionality, it sounds like it would be better fit as its own function that calls `isDefinitelyType` internally.

But I'm still not sure that this is necessary here. The `||` in the original code provides the short circuiting behavior you were talking about and is far more obvious because it's just an `||`.

Reply:
> If it is the case that we need this functionality, it sounds like it would be better fit as its own function that calls `isDefinitelyType` internally.

Yeah, I was thinking about that. I'll try it and see how that works.

> But I'm still not sure that this is necessary here. The `||` in the original code provides the short circuiting behavior you were talking about and is far more obvious because it's just an `||`.

It does not, because the left node is always checked and reported, even if this is not necessary because the right node has a type that matches.